### ISSUE TITLE
A11y no hover for uncompleted strings in Dropdown and OverflowMenu list

### ISSUE THREAD

**Issue by goldfeyesh** [carbon-design-system/carbon#6637]
Hi, I work on Key Protect UI team, and we've been having issues with Dropdown and OverflowMenu hover text not appearing on focused list items when using keyboard controls. I'm not well-versed in accessibility, so let me know if you need any clarifications on this - it was brought up to us by global shared services team.

## Environment

> Operating system

Mac OS 10.14.6
> Browser

Google Chrome 84.0.4147.105

> Automated testing tool and ruleset

https://www.ibm.com/able/guidelines/ci162/content_on_hover_71.html
> Assistive technology used to verify

Using tabs and arrow keys

## Detailed description

> What version of the Carbon Design System are you using?

"carbon-components": "10.16.0",
"carbon-components-react": "7.16.0",

> What did you expect to happen?

I can see the hover text message when I use mouse over these list items. I would expect this hover text to appear when I use keyboard controls to focus these list items.
<img width="417" alt="Screen Shot 2020-08-07 at 2 46 42 PM" src="https://user-images.githubusercontent.com/13471107/89683140-9c3ec780-d8bd-11ea-87c4-96360266a055.png">

<img width="352" alt="Screen Shot 2020-08-07 at 2 48 05 PM" src="https://user-images.githubusercontent.com/13471107/89683146-9d6ff480-d8bd-11ea-84a6-536518c01f04.png">

> What happened instead?

No hover text appears when I used arrow keys to focus these list items.
<img width="225" alt="Screen Shot 2020-08-07 at 2 47 31 PM" src="https://user-images.githubusercontent.com/13471107/89683143-9cd75e00-d8bd-11ea-8f4c-337c919eb8a1.png">
<img width="233" alt="Screen Shot 2020-08-07 at 2 48 38 PM" src="https://user-images.githubusercontent.com/13471107/89683148-9e088b00-d8bd-11ea-835f-6fadb78a2989.png">

> What WCAG 2.1 checkpoint does the issue violate?

https://www.ibm.com/able/guidelines/ci162/content_on_hover_71.html

## Steps to reproduce the issue

1. Use tab and enter or space to navigate to a Dropdown or OverflowMenu and open it.
2. Use arrow keys to focus a list item

This is present in the Carbon docs for these components too.
https://www.carbondesignsystem.com/components/dropdown/code/
https://www.carbondesignsystem.com/components/overflow-menu/code

**Comment by dakahn** [comment]
This is a good call out. Thanks for reporting. If we allow title attributes to be distinct from visible labels they should be keyboard accessible for sure and _specifically for this case_ having a keyboard accessible tooltip here is vital to the user understanding the element. 

As an aside, I don't much care for truncation in cases like this. Quintessential UI form over function that can add loads of dev and implementation complexity.
